File: backend/Domain/Models/Department.cs
namespace backend.Domain.Models
{
    public class Department : BaseEntity
    {
        private Department() { }

        public string DepartmentName { get; private set; }
        public string? Description { get; private set; }
        public bool? IsActive { get; private set; } = true;

        public Department(string departmentName, string? description)
        {
            DepartmentName = departmentName;
            Description = description ?? string.Empty;
        }

        public void Update(string? departmentName, string? description, bool? isActive)
        {
            DepartmentName = departmentName ?? DepartmentName;
            Description = description ?? Description;
            IsActive = isActive ?? IsActive;
        }
    }
}

File: frontend/src/types/department.ts
// ─── Department ──────────────────────────────────────────
export interface Department {
  id: number;
  departmentName: string;
  description: string | null;
  isActive: boolean | null;
  createdAt: string;
  isDeleted: boolean;
}

export interface CreateDepartmentRequest {
  departmentName: string;
  description?: string;
}

export interface UpdateDepartmentRequest {
  departmentName?: string;
  description?: string;
  isActive?: boolean;
}
